lib/metadata-json-lint/version_requirement.rb

```diff
@@ -47,6 +47,22 @@ def open_ended?
       end
     end
 
+    def puppet_eol?
+      true if range.begin.to_s =~ /^[2-4]\.[0-9]\.\d$/
```

Comment:
I really don't like a regex for version comparisons. Isn't there a real function to do comparison?

Reply:
To my knowledge there is not a function for comparing SemVer strings. Not that I've found anyway. Every lookup I've done on it ends up in even more complex loops and regex.

Reply:
`range.begin < SemanticPuppet::Version.parse(MIN_PUPPET_VER)` would work. In my test I actually moved the constant to `MetadataJsonLint::MIN_PUPPET_VER` so it wasn't a constant in the global scope.